File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        // Approach 1: Using set - O(n) & O(n)
        // set<int> s(nums.begin(), nums.end());
        // return s.size() != nums.size();

        // Approach 2: Sorting - O(n log n) & O(1)
        sort(nums.begin(), nums.end());

        for (int i = 1, n = nums.size(); i < n; i++) {
            if (nums[i] == nums[i - 1])
                return true;
        }
        return false;



    }
};

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        # Using sorting 
        # nums.sort()
        # for i in range(1, len(nums)):
        #     if nums[i] == nums[i - 1]:
        #         return nums[i]
        # return -1

        # Using hashmap
        hashmap = defaultdict(int)
        
        for num in nums:
            if num in hashmap:
                return num
            hashmap[num] = 1
        return -1
        